### DISCUSSION TITLE
issue: Rendered form state errors after trigger out of sync in tests with fakeTimers

### DISCUSSION THREAD

**Discussion by sbland** [react-hook-form/react-hook-form#8471] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/goofy-swanson-thy2mg?file=/src/app.spec.tsx

### Steps to reproduce

In `jest.config.js` set `fakeTimers: { enableGlobally: true }`;
Get the trigger as below: 
``` jsx
const methods = useForm<QuestionOutput>({
        resolver: yupResolver(validationSchema),
    });
const {
        trigger,
        formState: { errors },
    } = methods;
```

Call the trigger.
Wait for errors to render.
Full code in codesandbox: https://codesandbox.io/s/goofy-swanson-thy2mg?file=/src/app.spec.tsx
See `App.tsx` and `App.spec.tsx`

### Expected behaviour

When fake timers is off it successfully finds the rendered errors.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please refer to the example here for unit tests with `trigger`: https://github.com/react-hook-form/react-hook-form/blob/master/src/__tests__/useForm/trigger.test.tsx

**Reply by sbland** [reply]
Thanks @bluebill1049  I tried the first test at your link. It worked until I set in `jest.config.js` set `fakeTimers: { enableGlobally: true };` then the test fails at line 49. I tried adding `jest.runAllTimers();` to line 48 but this didn't change anything.
